java/src/humbug/Reporter.java

```diff
@@ -0,0 +1,180 @@
+package humbug;
+
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONObject;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+/**
+ * Reporter engine that will send reports to bugout.dev server
+ */
+public class Reporter {
+    private static String apiURL =  "https://spire.bugout.dev";
+
+    private final SystemInformation systemInformation;
+    private String name;
+    private HumbugConsent consent;
+    private String clientId;
+    private String sessionId;
+    private String bugoutToken;
+
+    public Reporter(String name, HumbugConsent consent, String clientId, String sessionId, String bugoutToken) {
+        this.systemInformation = SystemInformation.generateSystemInformation();
+        this.name = name;
+        this.consent = consent;
+        this.clientId = clientId;
+        this.sessionId = sessionId;
+        this.bugoutToken = bugoutToken;
```

Comment:
Add `tags` similar to what we added in Python version recently. These are tags that are associated with a reporter that get published with every report.